openoa/analysis/electrical_losses.py

```diff
@@ -3,290 +3,353 @@
+        self.process_scada()
+        if self.plant.metadata.meter.frequency not in ("MS", "M", "1MS"):
+            self.process_meter()
+            self.monthly_meter = False
 
     @logged_method_call
-    def run(
-        self, uncertainty_meter=0.005, uncertainty_scada=0.005, uncertainty_correction_thresh=0.95
-    ):
+    def run(self):
         """
-        Run the electrical loss calculation in order by calling this function.
-
-        Args:
-         uncertainty_meter(:obj:`float`): uncertainty imposed to revenue meter data (for UQ = True case)
-         uncertainty_scada(:obj:`float`): uncertainty imposed to scada data (for UQ = True case)
-         uncertainty_correction_thresh(:obj:`tuple`): Data availability thresholds (fractions)
-                                                         under which months should be eliminated.
-                                                         This should be a tuple in the UQ = True case,
-                                                         a single value when UQ = False.
-        Returns:
-            (None)
+        Run the electrical losses calculation.
         """
-        # Define uncertainties and check types
-        expected_type = float if not self.UQ else tuple
-        assert (
-            type(uncertainty_correction_thresh) == expected_type
-        ), f"uncertainty_correction_thresh must be {expected_type} for UQ={self.UQ}"
-
-        self.uncertainty_correction_thresh = np.array(
-            uncertainty_correction_thresh, dtype=np.float64
-        )
-        if self.UQ:
-            self.uncertainty_meter = uncertainty_meter
-            self.uncertainty_scada = uncertainty_scada
-
-        # Process SCADA data to daily sums
-        self.process_scada()
-
-        # Process meter data to daily sums (if time frequency is less than monthly)
-        self._monthly_meter = True  # Keep track of reported meter data frequency
-
-        if (
-            (self._plant._meter_freq != "MS")
-            & (self._plant._meter_freq != "M")
-            & (self._plant._meter_freq != "1MS")
-        ):
-            self.process_meter()
-            self._monthly_meter = False  # Set to false if sub-monthly frequency
-
-        # Setup Monte Carlo approach
+        # Setup Monte Carlo approach, and calculate the electrical losses
         self.setup_inputs()
-
-        # Calculate electrical losses, Monte Carlo approach
         self.calculate_electrical_losses()
 
     def setup_inputs(self):
         """
         Create and populate the data frame defining the simulation parameters.
-        This data frame is stored as self._inputs
-
-        Args:
-            (None)
-
-        Returns:
-            (None)
+        This data frame is stored as self.inputs.
         """
         if self.UQ:
+            n_decimal = (len(str(el.split(".")[1])) for el in self.uncertainty_correction_threshold)
+            integer_multiplier = 10**n_decimal
```

Comment:
I can't get this line to work. I receive the error "TypeError: unsupported operand type(s) for ** or pow(): 'int' and 'generator'" which seems to be caused by raising  integer 10 to a generator. The pow operator is only defined for numpy arrays.

Reply:
That's my fault, it's supposed to be `max(len...)`.